src/main/java/spiffe/api/svid/WorkloadAPIClient.java

```diff
@@ -0,0 +1,96 @@
+package spiffe.api.svid;
+
+import io.grpc.stub.StreamObserver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spiffe.api.svid.util.ExponentialBackOff;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static spiffe.api.svid.Workload.*;
+
+/**
+ * Provides functionality to interact with a Workload API
+ *
+ */
+public final class WorkloadAPIClient {
+
+    private static Logger LOGGER = LoggerFactory.getLogger(WorkloadAPIClient.class);
+
+    private SpiffeWorkloadStub spiffeWorkloadStub;
+
+    /**
+     * Constructor
+     * @param spiffeEndpointAddress
+     */
+    public WorkloadAPIClient(String spiffeEndpointAddress) {
+        spiffeWorkloadStub = new SpiffeWorkloadStub(spiffeEndpointAddress);
+    }
+
+    /**
+     * Default constructor
+     * The WorkloadAPI Address will be resolved by an Environment Variable
+     *
+     */
+    public WorkloadAPIClient() {
+        spiffeWorkloadStub = new SpiffeWorkloadStub(EMPTY);
+    }
+
+    /**
+     * Fetch the SVIDs from the Workload API on a synchronous fashion
+     * Use a Exponential Backoff to handle the errors and retries
+     *
+     * @return List of X509SVID or Empty List if none have been fetched
+     */
+    public List<X509SVID> fetchX509SVIDs() {
```

Comment:
I am wondering if we should provide a synchronous API at all... If folks use this API, they are likely to screw up rotation.

Reply:
Removed the synchronous call